plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java

```diff
@@ -107,6 +109,54 @@ public void setEvent(CalendarEvent calendarEvent) {
 		event.setReminders(calendarEvent.getReminders());
 		this.calendarService.addEvent("primary", event);
 	}
+	
+	/**
+	 * @param entities
+	 * 			from the speech
+	 * @return if the event was successfully created or if not why
+	 */
+	public String createNewEvent(Map<String, EntityData> entities) {
+		boolean allDay = isAllDay(entities.get("allday").getString());
+		
+		boolean startBeforeEnd = true;
+		LocalDateTime start;
+		LocalDateTime end;
+		if(allDay) {
+			start = LocalDateTime.of(entities.get("startdate").getDate().plusDays(1), this.zero);
+			end = LocalDateTime.of(entities.get("enddate").getDate().plusDays(2), this.zero);
+			startBeforeEnd = start.isBefore(end) || start.isEqual(end);	
+		} else if(entities.get("starttime") == null || entities.get("endtime") == null) {
+			return "You have to restart the creation of a new event and please make sure that you add a time to the "
+					+ "start and to the end if you choose an non all day event.";
+		} else {
+    		start = LocalDateTime.of(entities.get("startdate").getDate(), entities.get("starttime").getTime());
+    		end = LocalDateTime.of(entities.get("enddate").getDate(), entities.get("endtime").getTime());
+    		startBeforeEnd = start.isBefore(end);	
+		}
+		if (!startBeforeEnd) {
+			return  "You have to restart the creation of a new event and please make sure that the start of the event "
+					+ "is before the end.";
+		}
+		
+		String location = entities.get("location").getString();
+		if (location.equals("no")) { location = ""; }
+		String description = entities.get("description").getString();
+		if (description.equals("no")) { description = ""; }
+		
+		int reminderTime = entities.get("remindertimevalue").getNumber();
+		String reminderTimeUnit = entities.get("remindertimeunit").getString();
+		if (reminderTimeUnit.equals("hours") || reminderTimeUnit.equals("h")) {
```

Comment:
is problematic if the user says hour you get at moment the user input and not the matched grammar word